pkg/controller/datavolume/clone-controller.go

```diff
@@ -65,6 +65,8 @@ const (
 	CloneInProgress = "CloneInProgress"
 	// SnapshotForSmartCloneInProgress provides a const to indicate snapshot creation for smart-clone is in progress
 	SnapshotForSmartCloneInProgress = "SnapshotForSmartCloneInProgress"
+	// CloneFromSnapshotSourceInProgress provides a const to indicate clone from snapshot source is in progress
```

Comment:
I won't get to into the nuts an bolts of this implementation until rebased on #2525 but I'm wondering if PVC source and snapshot source should be two different controllers?  May be more readable than having everything in this file?

Reply:
Yeah I was holding back the rebase push until the review so I don't throw anyone off, my thinking is:

- Lots of shared functionality with clone => need to make a clone controller template to reuse => not sure it's worth the overcomplication when snapshots should be the last "clone" source we add
(we go all PVCs soon, people will be doing restores themselves, otherwise, they'll use host assisted populator?)
- Path distinction between PVC & Snapshot source happens very early on so not too horrible?

Reply:
Totally understand that there are many shared operations between pvc and snapshot source.  My gut tells me that snapshot should be a dedicated controller.  What do you think @arnongilboa?

Reply:
I tend to think snapshot should have a dedicated controller, and both should have a common base (like the current DV controllers). Since it introduces lots of new code, maintenance will be much easier this way. We should learn the lesson from the huge DV controller spagetti dish we had before the split/refactoring.

Reply:
Yep totally makes sense!
Let's hop down the bunny trail over to https://github.com/kubevirt/containerized-data-importer/pull/2544

Reply:
@mhenriks @arnongilboa implementation with separate controller ready for review